cases/src/org/commcare/cases/util/CasePurgeFilter.java

```diff
@@ -132,61 +126,91 @@ private void setIdsToRemoveNew(IStorageUtilityIndexed<Case> caseStorage, Vector<
             indexHolder.removeAllElements();
         }
 
-        //Pass 2: Start at Sources (nodes which only produce edges) and walk up
-        //the tree, marking relevant nodes as alive
-        Stack<String> toProcess = g.getSources();
-        while (!toProcess.isEmpty()) {
-            //current node
-            String index = toProcess.pop();
-            String[] node = g.getNode(index);
+        markRelevant(g);
+        markAvailable(g);
+        markLive(g);
+    
+        //Ok, so now just go through all nodes and signal that we need to remove anything
+        //that isn't live!
+        for (Enumeration<int[]> iterator = g.getNodes(); iterator.hasMoreElements(); ) {
```

Comment:
This and all other instances of `for (Enumeration<SomeType> ...` will cause J2ME builds to fail, right?